### PR TITLE
feat(policy): Add Rego policy engine package

### PR DESCRIPTION
**Pull request by jiparis**
This PR adds a new package to evaluate policies, and a default implementation using OPA Rego language. 

Note that this feature is still not connected to Chainloop contracts, but I wanted to send an early PR to make it easier to review.

Refs #122

### CODE REVIEW

File: pkg/policies/engine/engine.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	Verify(ctx context.Context, policy *Policy, input []byte) ([]*PolicyViolation, error)
+}
+
+type PolicyViolation struct {
+	Subject, Violation string
+}
+
+// Policy represents a loaded policy in any of the supported technologies.
+type Policy struct {
+	//
```
Comment (javirln) [inline pkg/policies/engine/engine.go:32]: Missing comment?

File: pkg/policies/engine/engine.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	Module []byte `json:"module"`
+	Name   string `json:"name"`
```
Comment (javirln) [inline pkg/policies/engine/engine.go:34]: Could you explain what those two would mean?

File: pkg/policies/engine/rego/rego.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		return nil, fmt.Errorf("failed to parse input: %w", err)
+	}
+
+	// add input
+	regoInput := rego.Input(decodedInput)
+
+	// add module
+	regoFunc := rego.ParsedModule(parsedModule)
+
+	// add query. Note that the predefined rule to look for is `deny`
```
Comment (javirln) [inline pkg/policies/engine/rego/rego.go:57]: Shall we document this?

**Review by javirln — APPROVED** [review]
Added small comments but, looking awesome :D

File: pkg/policies/engine/rego/rego.go (hunk 3 above)
Comment (jiparis) [reply pkg/policies/engine/rego/rego.go:57]: Yes, I plan to add documentation once we release this feature.

**Pull request merged**